util.go

```diff
@@ -24,6 +24,10 @@ const (
 	SnowflakeRequestIDKey contextKey = "SNOWFLAKE_REQUEST_ID"
 	// streamChunkDownload determines whether to use a stream based chunk downloader
 	streamChunkDownload paramKey = "STREAM_CHUNK_DOWNLOAD"
+	//FetchResultByID the queryID of query result to fetch
+	FetchResultByID paramKey = "SF_FETCH_RESULT_BY_ID"
+	// debugGetFunc carry debug func to override internal GET
+	debugCustomFunc paramKey = "DEBUG_GET_FUNC"
```

Comment:
there should be a cleaner way to do this, or to intercept by using a HTTP round tripper transport defined in the config